gadget/ondisk.go

```diff
@@ -255,12 +248,11 @@ func deviceName(name string, index int) string {
 // returns a partitioning description suitable for sfdisk input and a
 // list of the partitions to be created.
 func BuildPartitionList(dl *OnDiskVolume, pv *LaidOutVolume) (sfdiskInput *bytes.Buffer, toBeCreated []OnDiskStructure) {
-	ptable := dl.partitionTable
-
 	// Keep track what partitions we already have on disk
-	seen := map[uint64]bool{}
-	for _, p := range ptable.Partitions {
-		seen[p.Start] = true
+	seen := map[quantity.Size]bool{}
```

Comment:
in retrospect, maybe we should have added quantity.Offset too, just to make it explicit that it is an offset rather than size

Reply:
we can still add it :) when we have a moment (naive grepping gives around ~100 places to change)

Reply:
Yeah, something I'll take a look at today.

Reply:
This is done in #9781